framework/include/gpu/base/GPUMatrix.h

```diff
@@ -0,0 +1,106 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GPUArray.h"
+
+#include "libmesh/petsc_matrix.h"
+
+class GPUSystem;
+
+class GPUMatrix
```

Comment:
Class docstring. Move function definitions outside of class declaration. Needs to e namespaced